File: WowUp.Common/Models/Warcraft/ProductDb.cs
using ProtoBuf;

namespace WowUp.Common.Models.Warcraft
{
    [ProtoContract]
    public class ProductDb
    {
        [ProtoMember(1)]
        public Product[] Products { get; set; }

        [ProtoMember(7)]
        public string[] ProductNames { get; set; }
    }
}

File: wowup-electron/src/app/models/warcraft/product-db.ts
import { Field, Message, Type } from "protobufjs";
import { Product } from "./product";

@Type.d("ProductDb")
export class ProductDb extends Message<ProductDb> {
  @Field.d(1, Product, "repeated")
  public products: Product[];

  @Field.d(7, "string", "repeated")
  public productNames: string[];
}
